lib/backend/etcdbk/etcd_test.go

```diff
@@ -366,3 +367,31 @@ func (s *EtcdSuite) TestSyncLegacyPrefix(c *check.C) {
 	})
 	s.reset(c)
 }
+
+// CompareAndSwapIssue4786 ensures that the backend reacts with a proper
+// error message if client sends a message exceeding the configured size maximum
+// See https://github.com/gravitational/teleport/issues/4786
+func (s *EtcdSuite) TestCompareAndSwapOversizedValue(c *check.C) {
+	// setup
+	const maxClientMsgSize = 128
+	bk, err := New(context.Background(), backend.Params{
+		"peers":                     []string{"https://127.0.0.1:2379"},
+		"prefix":                    "/teleport",
+		"tls_key_file":              "../../../examples/etcd/certs/client-key.pem",
+		"tls_cert_file":             "../../../examples/etcd/certs/client-cert.pem",
+		"tls_ca_file":               "../../../examples/etcd/certs/ca-cert.pem",
+		"dial_timeout":              500 * time.Millisecond,
+		"max_client_msg_size_bytes": maxClientMsgSize,
+	})
+	c.Assert(err, check.IsNil)
+	prefix := test.MakePrefix()
+	// Explicitly exceed the message size
+	value := make([]byte, maxClientMsgSize+1)
+
+	// verify
+	_, err = bk.CompareAndSwap(context.TODO(),
```

Comment:
Nit: Any reason you used `context.Background()` above but `context.TODO()` here? :)

Reply:
I did not choose Background (above) but my reason for TODO is that `Background` is usually reserved for combinations like `WithCancel` or `WithTimeout` while `TODO` is a clear indication that a proper context should be used here but it is undecided how to handle the cancellation/timeouts yet.